### ISSUE TITLE
[Bug]: expect.poll fails with custom timeout when debug

### ISSUE THREAD

**Issue by Diokuz** [microsoft/playwright#31501]
### Version

1.45.0

### Steps to reproduce

1. Run test with `--debug`
2. Click `Step over (F10)` every 2000 ms
3. expect.poll fails because it has shorter timeout

```
      await expect
        .poll(
          async () => this.page.evaluate(() => window.STORYBOOK_RENDERER),
          {
            message: '...',
            timeout: 1000,
            intervals: [10, 100, 400, 1000],
          },
        )
        .toBeUndefined()
```

### Expected behavior

expect.poll counts only for non-user-interactive time

### Actual behavior

```
Timeout 1000ms exceeded while waiting on the predicate
```

### Environment

```shell
Mac Os X


node -v
v18.12.1
```
```

**Comment by Diokuz** [comment]
Same for the default 5000 ms timeout

```
Timeout 5000ms exceeded while waiting on the predicate
```

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
